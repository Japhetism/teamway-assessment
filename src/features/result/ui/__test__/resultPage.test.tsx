import { render, screen, within } from "@testing-library/react";
import { createRenderer } from "react-test-renderer/shallow";
import Result from "..";

const renderer = createRenderer();

const defaultComponent = <Result />;

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Unit test for result page ", () => {

    it("should render and match the snapshot", () => {
        renderer.render(defaultComponent);
        const renderedOutput = renderer.getRenderOutput();

        expect(renderedOutput).toMatchSnapshot();
    });

    it("should have title as Your result", () => {
        const { getByRole } = render(defaultComponent);
        const titleHeading = getByRole("heading", { name: "Your Result" });

        expect(titleHeading).toBeVisible();
    });

});
