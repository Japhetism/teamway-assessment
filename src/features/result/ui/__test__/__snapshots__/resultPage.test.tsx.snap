// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for result page  should render and match the snapshot 1`] = `
<div
  className="w-3/5 bg-white p-10"
>
  <div
    className="text-center"
  >
    <h5>
      Your Result
    </h5>
    <h3
      className="text-4xl mt-5"
    >
      You are more of an 
    </h3>
  </div>
  <div
    className="mt-10"
  >
    <p>
      
    </p>
  </div>
</div>
`;
